### PR TITLE
ci: (experimental) Enable more ruff rules

### PR DESCRIPTION
**Pull request by aahung**
### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ssenchenko — APPROVED** [review]

**Review by GavinZZ — APPROVED** [review]

File: samtranslator/intrinsics/actions.py [hunk 1]
```diff
@@ -505,7 +512,7 @@ def resolve_resource_id_refs(
     def _check_input_value(self, value: Any) -> bool:
         # Value must be an array with *at least* two elements. If not, this is invalid GetAtt syntax. We just pass along
```
Comment (acristin) [inline samtranslator/intrinsics/actions.py:513]: nit, `*at least* _MIN_NUM_ARGUMENTS elements`

File: samtranslator/model/intrinsics.py [hunk 2]
```diff
@@ -88,7 +92,7 @@ def make_combined_condition(
     :param string condition_name: base name desired for new condition
     :return: dictionary of condition_name: condition_value
     """
-    if len(conditions_list) < 2:
+    if len(conditions_list) < MIN_NUM_CONDITIONS_TO_COMBINE:
         # Can't make a condition if <2 conditions provided.
```
Comment (acristin) [inline samtranslator/model/intrinsics.py:96]: Similar nit about hard coded number in comment.

File: samtranslator/model/sam_resources.py [hunk 3]
```diff
@@ -456,8 +459,8 @@ def _make_gen_condition_name(self, name: str, hash_input: str) -> str:
         # Make sure the property name is not over 255 characters (CFN limit)
```
Comment (acristin) [inline samtranslator/model/sam_resources.py:459]: Nit: Could reword as "is not over the CFN limit (currently 255)". I doubt this limit will ever change, so this is probably not a big deal.

File: samtranslator/third_party/py27hash/hash.py [hunk 4]
```diff
@@ -104,7 +106,7 @@ def fhash(value):  # type: ignore[no-untyped-def]
         """
 
         fpart = math.modf(value)
-        if fpart[0] == 0.0:
+        if fpart[0] == Hash._FLOAT_ZERO:
```
Comment (acristin) [inline samtranslator/third_party/py27hash/hash.py:109]: Why is this prefixed by `Hash.`? Shouldn't `_FLOAT_ZERO` suffice?

Comment (aahung) [reply samtranslator/third_party/py27hash/hash.py:109]: Its scope is still inside the class.

File: samtranslator/intrinsics/actions.py (hunk 1 above)
Comment (aahung) [reply samtranslator/intrinsics/actions.py:513]: updated to "enough"

File: samtranslator/model/intrinsics.py (hunk 2 above)
Comment (aahung) [reply samtranslator/model/intrinsics.py:96]: updated to "enough"

File: samtranslator/model/sam_resources.py (hunk 3 above)
Comment (aahung) [reply samtranslator/model/sam_resources.py:459]: updated

**Review by acristin — APPROVED** [review]

**Pull request merged**